spec/features/consumer/shopping/embedded_shopfronts_spec.rb

```diff
@@ -110,8 +111,8 @@
   # When we click the "Update" button to save changes, we need to wait
   #   until it is actually saved and "loading" disappears before doing anything else.
   def wait_for_shop_loaded
-    page.has_no_content? "Loading"
-    page.has_no_css? "input[value='Updating cart...']"
```

Comment:
We're clearly missing something. I thought these methods wait for the text/css to disappear.

Reply:
I always struggle to find information about this. https://blog.codeship.com/faster-rails-tests/ suggests that `has_no_content?` is waiting. I just tested it:

- `page.has_content?` is waiting
- `page.has_no_content?` is waiting
- `page.has_no_css?` is waiting

[This post](https://stackoverflow.com/a/36842392/3377535) says that `has_content?` is just a wrapper for `have_content`.

@Matt-Yorkley How did you test that it doesn't block? There may be something else we need to wait for and your change made the failure only less likely.

Reply:
Yes, I also remember that article talking about it. Actually, I didn't know before.

Reply:
See https://github.com/openfoodfoundation/openfoodnetwork/pull/3339#issuecomment-455655974